### DISCUSSION TITLE
Option for Timeseries Null days to be shown

### DISCUSSION THREAD

**Discussion by alanmcruickshank** [apache/superset#18379] [Ideas]
New Feature Request:

For timeseries plots, at the moment if there is a day with no data, the day is skipped and the day isn't shown at all. I'd love an option where these days could be shown even though they're blank, so that there would be an intentional gap.

Something similar to the "Show Missing Values" option available in packages like Tableau: https://onlinehelp.tableau.com/current/pro/online/mac/en-us/missing_values.html

**Comment by LAlbertalli** [comment]
Any updates on these feature request? It is really annoying, especially for datasets with low-frequency data.

**Comment by alanmcruickshank** [comment]
@LAlbertalli 
This is currently possible through using the advanced options section and re-sampling using the re-sample rule which corresponds to the relevant time grain. 

The UX is a bit rough because if you change the time grain you also have to change the resample rule. What would be excellent is a way of tying a resample rule to a time grain so they both change together - perhaps as an inbuilt function within certain visualisations.

**Comment by mistercrunch** [comment]
There could be some auto-resampling based on Grain, but it would require tighter semantics around the way Grains are defined. Currently `Grains` are defined by database dialects and they're pretty loose list of names and SQL expressions. I guess we'd only have to add a 3rd configuration element as a Pandas `freq` expression (`7D`, `12H`, ...).

I think it'd be pretty trivial actually. If that pandas `freq` is defined for that grain, we'd call the pandas resampling function automatically.

**Comment by benvogan** [comment]
@alanmcruickshank Unless I am missing something the pandas resampling methods either backfill or forward-fill.  There is no fill method (that I can find) that zero fills - which is what we really want.

**Comment by mistercrunch** [comment]
I think `fill_method=None` will fill with NA, which translates to gaps in trendlines in nvd3 as opposed to a non-linear axis. It's easy to call `fillna(0)` if you want to replace those NA too. Though users might want either... I'd say default should be NA.

**Comment by mistercrunch** [comment]
Notice: this issue has been closed because it has been inactive for 383 days. Feel free to comment and request for this issue to be reopened.

**Comment by trtg** [comment]
It would be ideal to have some way to show gaps in data, since that can often indicate some sort of system failure (broken ingestion, etc). The current approach of effectively interpolating can mask problems. Setting the How dropdown of of the pandas.resample section to sum, the Rule dropdown to whatever your interval is and leaving the Fill Method dropdown blank, seems to do what I want in this particular case. @mistercrunch @benvogan

**Comment by timnick-snow** [comment]
Any updates on these feature request?

**Comment by villebro** [comment]
This is currently supported in the new timeseries charts (e.g. "Time-series Line Chart"). Just expand advanced analytics, select the expected frequency and fill method in the "Resample" section and you'll get filled in values:
<img width="1071" alt="image" src="https://user-images.githubusercontent.com/33317356/154445637-b2c1bee5-d27e-4440-a2ab-82eaa7d36641.png">

@zhaoyongjie it would be cool to add an "autodetect" or "Time grain" option to default to infer the frequency based on the grain. Thoughts?

**Reply by inctl** [reply]
Yes, the grain isn't dynamic.
The best solution would be to offer this feature in conjunction with the time grain selection.
Without that, we can't provide graphs with dynamic grain.

**Reply by rusackas** [reply]
Agreed, but let's start a new discussion rather than resurrecting this decade old one that DID fix the original complaint. 

PRs/discussions welcome, but trying to keep the context manageable ;)

**Reply by inctl** [reply]
Yes @rusackas 
I create this issue for that : [42587](https://github.com/apache/superset/issues/42587).
